### ISSUE TITLE
Build failed in both VS Code and Android Studio

### ISSUE THREAD

**Issue by hayschan** [thingsboard/flutter_thingsboard_pe_app#59]
### Environment
- Flutter Version: `3.18.0-18.0.pre.47`
- Dart Version: `3.3.0 (build 3.3.0-273.0.dev)`
- Operating System: Windows 11
- Android Studio version: `2022.3`
- VS Code version: `1.85.1`, Flutter extension version `3.80.0`

### Description
I am a software engineer working on the Flutter ThingsBoard PE Mobile Application, and I am encountering an issue with building the app. Despite following the official documentation's getting-started guide, the build process fails in both VS Code and Android Studio.

### Steps to Reproduce
1. Cloned the repository from https://github.com/thingsboard/flutter_thingsboard_pe_app
2. Followed the setup instructions from https://thingsboard.io/docs/pe/mobile/getting-started/. For the code of the PE app, I have changed nothing but `thingsBoardApiEndpoint = 'https://thingsboard.cloud';`
3. Attempted to build the app using VS Code. VS code launched the emulator, but hung on `Flutter: Launching ...` forever - received the following log: 
    ```python
    Launching lib\main.dart on sdk gphone64 x86 64 in debug mode...
    ```
5. Attempted to build the app using Android Studio - received a similar error:
    ```python
    Launching lib\main.dart on sdk gphone64 x86 64 in debug mode...
    Running Gradle task 'assembleDebug'...
    Exception in thread "main" java.lang.RuntimeException: Timeout of 120000 reached waiting for exclusive access to file: C:\Users\hayschan\.gradle\wrapper\dists\gradle-7.5.1-all\1ehga6e77gqps5uk2kc5kf1vc\gradle-7.5.1-all.zip
	    at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:61)
	    at org.gradle.wrapper.Install.createDist(Install.java:48)
	    at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:128)
	    at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
    Retrying Gradle Build: #1, wait time: 100ms
    [!] Gradle threw an error while downloading artifacts from the network.
    Running Gradle task 'assembleDebug'...
    Exception in thread "main" java.lang.RuntimeException: Timeout of 120000 reached waiting for exclusive access to file: C:\Users\hayschan\.gradle\wrapper\dists\gradle-7.5.1-all\1ehga6e77gqps5uk2kc5kf1vc\gradle-7.5.1-all.zip
	    at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:61)
	    at org.gradle.wrapper.Install.createDist(Install.java:48)
	    at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:128)
	    at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
    [!] Gradle threw an error while downloading artifacts from the network.
    Error: Gradle task assembleDebug failed with exit code 1

    ```

### Expected Behavior
The app should build successfully following the steps provided in the official documentation.

### Additional Context
- I am a paid customer on the Prototype plan of Thingsboard Cloud.

I would appreciate any guidance or solutions to resolve this build issue. Thank you for your assistance.

Best regards,
Hays Chan

**Comment by hayschan** [comment]
## More context

I managed to build the app using the command `flutter run`. I got the following error:

```shell
> flutter run
Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source!
Launching lib\main.dart on sdk gphone64 x86 64 in debug mode...
You are applying Flutter's app_plugin_loader Gradle plugin imperatively using the apply script method, which is deprecated and will be removed in a future release. Migrate to applying Gradle plugins with the declarative plugins block: https://flutter.dev/go/flutter-gradle-plugin-apply

You are applying Flutter's main Gradle plugin imperatively using the apply script method, which is deprecated and will be removed in a future release. Migrate to applying Gradle plugins with the declarative plugins block: https://flutter.dev/go/flutter-gradle-plugin-apply


FAILURE: Build failed with an exception.

* Where:
Script 'D:\GitHub\flutter\packages\flutter_tools\gradle\src\main\groovy\flutter.groovy' line: 1178

* What went wrong:
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Could not create task ':app:copyFlutterAssetsDebug'.
      > Could not create task ':app:mergeDebugAssets'.
         > Cannot use @TaskAction annotation on method IncrementalTask.taskAction$gradle_core() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 1s
Running Gradle task 'assembleDebug'...                           2,485ms
Error: Gradle task assembleDebug failed with exit code 1
```

**Comment by hayschan** [comment]
## TL;DR

To resolve the build issue in the Flutter ThingsBoard PE Mobile Application, the following changes were made:

1. **`build.gradle`**: Added three local maven URLs to the `repositories` section in both the `buildscript` and `allprojects` blocks, replacing the original `google()` and `jcenter()` sources.

2. **`gradle-wrapper.properties`**: Changed the `distributionUrl` from the standard gradle.org URL to a local mirror provided by a Chinese internet company.

These modifications addressed network issues specific to China, likely caused by the Great Firewall (GFW), and allowed successful build and emulation of the app.

## My solution

The build is successful now. I can see it running on the emulator.

I am located in China. The network here is special due to the GFW. Modifications have to be done to two files:

### `build.gradle`

I added three local maven URL in two places in this file.

```python
buildscript {
    ext.kotlin_version = '1.7.10'
    repositories {
        maven { url 'https://maven.aliyun.com/repository/google' }
        maven { url 'https://maven.aliyun.com/repository/jcenter' }
        maven { url 'https://maven.aliyun.com/nexus/content/groups/public' }
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:7.2.2'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
    }
}

allprojects {
    repositories {
        maven { url 'https://maven.aliyun.com/repository/google' }
        maven { url 'https://maven.aliyun.com/repository/jcenter' }
        maven { url 'https://maven.aliyun.com/nexus/content/groups/public' }
    }
}

rootProject.buildDir = '../build'
subprojects {
    project.buildDir = "${rootProject.buildDir}/${project.name}"
}
subprojects {
    project.evaluationDependsOn(':app')
}

tasks.register("clean", Delete) {
    delete rootProject.buildDir
}
```

### `gradle-wrapper.properties`

I changed the distribution URL from gradle.org to a local internet company's gradle mirror path.

```
#Fri Jun 23 08:50:38 CEST 2017
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
# distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-all.zip
distributionUrl=https\://mirrors.cloud.tencent.com/gradle/gradle-7.5.1-all.zip
```
